wicketd-client/src/lib.rs

```diff
@@ -38,9 +38,14 @@ progenitor::generate_api!(
         RotInventory = { derives = [ PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize]},
         RotSlot = { derives = [ PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize]},
         ImageVersion = { derives = [ PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize]},
+        UpdateComponent = { derives = [ Copy, PartialEq, Eq, PartialOrd, Ord ] },
     }
 );
 
+pub type ProgressEventKind = types::ProgressEventKindForWicketdEngineSpec;
+pub type StepEventKind = types::StepEventKindForWicketdEngineSpec;
+pub type StepInfoWithMetadata = types::StepInfoWithMetadataForWicketdEngineSpec;
```

Comment:
Ah interesting, the approach I took in #2830 was to instead depend on `update-engine` and use the `replace` directive. I think that approach works out slightly better overall because you can use methods defined on the various types more easily.

If we go down this path, we'd need to split some of the code out into a separate `wicketd-types` or `wicketd-common` crate.

Reply:
TBH these were just to shorten all the type names - I didn't need any methods defined on the types in `update-engine`, but since there's (currently?) only one spec involved in wicketd it didn't seem necessary to have `..ForWicketdEngineSpec` attached to all the type names.

I think this falls into the "do we even need `replace`" which was some feedback from Adam at one point, I think? On this PR I don't think we do.

Reply:
Yeah I agree that this isn't immediately useful. But I'm wondering if reconciling this with #2830 isn't going to involve some of that. I think we can tackle that as needed though (I plan to do the integration work between the two.)